app/services/digest_exporter.rb

```diff
@@ -46,36 +47,38 @@ def digest_ids
     @digest_ids ||= initialize_digest_ids
   end
 
+  def column_headers
+    ApplicationDigest.column_headers
+  end
+
   def initialize_digest_ids
     log_message "Getting digest ids"
     ApplicationDigest.order(:created_at).pluck(:id)
   end
 
   def save_digest_to_rows(digest_ids)
     log_message "save_digest_to_rows"
+    log_message "Adding #{column_headers.count + 1} column headers, ApplicationDigest.column_headers + extraction date, to @rows"
+    @rows << [column_headers + extraction_date].flatten
     log_message "#{digest_ids.size} records to be added to @rows"
     digest_ids.each_with_index do |digest_id, _index|
       digest = ApplicationDigest.find(digest_id)
       @rows << digest.to_google_sheet_row
     end
     log_message "@row.count: #{@rows.size}"
-    log_message "@worksheet.rows.count: #{@worksheet.rows.count}"
+    log_message "@worksheet.max_rows: #{@worksheet.max_rows}"
   end
 
   def reset_worksheet
-    log_message "reset_worksheet"
-    # You can't delete all the rows in a worksheet, so we delete all but one, and we'll overwrite that
-    # with the column headers
-    @worksheet.delete_rows(1, @worksheet.max_rows - 1)
-    log_message "Existing data removed from spreadsheet"
+    return if @worksheet.max_rows == 1 && @worksheet.max_cols == 1
 
-    log_message "Writing #{ApplicationDigest.column_headers.count} column headers + extraction date"
-    # Overwrite the 1 remaining row with column headers
-    @worksheet.update_cells(1, 1, [ApplicationDigest.column_headers + extraction_date])
-    @worksheet.save
+    log_message "reset_worksheet"
+    clear_sheet("ROWS") unless @worksheet.max_rows == 1
+    clear_sheet("COLUMNS") unless @worksheet.max_cols == 1
+    @worksheet.reload
```

Comment:
Should this be `.save` instead of `.reload`

Reply:
No. Because we use the batch_update of google api calls in `clear_sheet` these are saved automatically *but* the cached version of @worksheet does not get updated without a reload :+1: